Update the connector API design (#16)
wso2-ballerina/module-amazons3: amazons3/API_Design.md

## Overview

> The Amazon S3 connector allows you to access the Amazon S3 REST API using ballerina. This connector has a base connector called "AmazonS3Cient" which exposes the commonly used functios such as create, get and delete buckets and objects. And other functions related to buckets and objects will be exposed through sub connectors(AmazonS3BucketClient and AmazonS3ObjectClient) for each other caegoriztion.

Reviewer: ```suggestion
The Amazon S3 connector allows you to access the Amazon S3 REST API using ballerina. This connector has a base connector called `AmazonS3Cient` which exposes the commonly used functions such as create, get and delete buckets and objects. And other functions related to buckets and objects will be exposed through sub connectors(`AmazonS3BucketClient` and `AmazonS3ObjectClient`) for each other categorization.
```

## API design for AmazonS3Client functions
### Functions

#### Create bucket
The createBucket function creates a new bucket.
 
##### Function signature

>  public remote function createBucket(string bucketName, CannedACL? cannedACL = ()) returns boolean|error;

Reviewer: ```suggestion
 `public remote function createBucket(string bucketName, CannedACL? cannedACL = ()) returns boolean|error;`
```

##### Parameters

|     Name               |    Type     |  Description   |
|:------------------:|:--------------:|:--------------:|
| bucketName        |   string         |    Unique name for the bucket to create.   |
|cannedACL | CannedACL | The access control list of the new bucket. |

##### Returns
    On success: Boolean value(true).
    On failure: error.
        
#### List buckets

> The listBuckets function retrieves a list of all Amazon S3 buckets that the authenticated user of the request owns. 

Reviewer: ```suggestion
The `listBuckets()` function retrieves a list of all Amazon S3 buckets that the authenticated user of the request owns. 
```
